test/jetstream/api/stream_test.exs

```diff
@@ -181,4 +199,92 @@ defmodule Jetstream.API.StreamTest do
       assert :ok = Stream.delete(:gnat, "PURGE_TEST")
     end
   end
+
+  describe "get_all_messages/2" do
+    setup do
+      on_exit(fn ->
+        nil
+      end)
+
+      :ok
+    end
+
+    test "gets multiple messages" do
+      stream = %Stream{
+        name: "GET_MULTIPLE_MESSAGES",
+        subjects: ["GET_MULTIPLE_MESSAGES.*"]
+      }
+
+      assert {:ok, _response} = Stream.create(:gnat, stream)
+
+      assert :ok = Gnat.pub(:gnat, "GET_MULTIPLE_MESSAGES.bar", "bar message")
+      assert :ok = Gnat.pub(:gnat, "GET_MULTIPLE_MESSAGES.foo", "foo message")
+
+      assert {:ok, ["bar message", "foo message"]} ==
+               Stream.get_all_messages(:gnat, "GET_MULTIPLE_MESSAGES")
+
+      Stream.delete(:gnat, "GET_MULTIPLE_MESSAGES")
+    end
+
+    test "when no messages" do
+      stream = %Stream{
+        name: "GET_MULTIPLE_MESSAGES",
+        subjects: ["GET_MULTIPLE_MESSAGES.*"]
+      }
+
+      assert {:ok, _response} = Stream.create(:gnat, stream)
+
+      assert {:ok, []} ==
+               Stream.get_all_messages(:gnat, "GET_MULTIPLE_MESSAGES")
+
+      Stream.delete(:gnat, "GET_MULTIPLE_MESSAGES")
+    end
+
+    min_server_version("2.7.2")
+
+    test "filter subject" do
+      stream = %Stream{
+        name: "GET_MULTIPLE_MESSAGES",
+        subjects: ["GET_MULTIPLE_MESSAGES.*"]
+      }
+
+      assert {:ok, _response} = Stream.create(:gnat, stream)
+
+      assert :ok = Gnat.pub(:gnat, "GET_MULTIPLE_MESSAGES.bar", "bar message")
+      assert :ok = Gnat.pub(:gnat, "GET_MULTIPLE_MESSAGES.foo", "foo message")
+
+      assert {:ok, ["foo message"]} ==
+               Stream.get_all_messages(:gnat, "GET_MULTIPLE_MESSAGES",
+                 filter_subject: "GET_MULTIPLE_MESSAGES.foo"
+               )
+
+      Stream.delete(:gnat, "GET_MULTIPLE_MESSAGES")
+    end
+
+    min_server_version("2.7.2")
```

Comment:
This look weird, like a lost function that once was in a test. It would look better if it was a normal tag like `@tag min_gnat_version: "2.7.2"`. Then, according to [this post](https://elixirforum.com/t/is-it-possible-to-mark-tests-as-incomplete-or-skipped-in-exunit/9400/9) it should be possible to skip this test from inside a setup in `conn_case`